project/BuildHelper.scala

```diff
@@ -7,65 +7,71 @@ import sbtbuildinfo._
 import BuildInfoKeys._
 
 object BuildHelper {
-  val testDeps        = Seq("org.scalatest"   %% "scalatest"    % "3.0.8" % "test")
-  val compileOnlyDeps = Seq("com.github.ghik" %% "silencer-lib" % "1.4.2" % "provided")
-
   val zioVersion = "1.0.0-RC14"
 
-  private val stdOptions = Seq(
-    "-deprecation",
-    "-encoding",
-    "UTF-8",
-    "-feature",
-    "-unchecked"
-  )
-
-  private val std2xOptions = Seq(
-    //"-Ymacro-debug-lite",
-    //"-Ymacro-debug-verbose",
-    "-Xfatal-warnings",
-    "-language:higherKinds",
-    "-language:existentials",
-    "-explaintypes",
-    "-Yrangepos",
-    "-Xsource:2.13",
-    "-Xlint:_,-type-parameter-shadow",
-    "-Ywarn-numeric-widen",
-    "-Ywarn-value-discard"
+  private val testDeps = Seq(
+    "org.scalatest" %% "scalatest" % "3.0.8" % "test"
   )
 
-  val buildInfoSettings = Seq(
-    buildInfoKeys := Seq[BuildInfoKey](name, version, scalaVersion, sbtVersion, isSnapshot),
-    buildInfoPackage := "zio",
-    buildInfoObject := "BuildInfoZioMacros"
-  )
-
-  val optimizerOptions = {
-    Seq(
-      "-opt:l:inline",
-      "-opt-inline-from:zio.internal.**"
+  private def compileOnlyDeps(scalaVersion: String) = {
+    val stdCompileOnlyDeps = Seq(
+      "com.github.ghik" %% "silencer-lib" % "1.4.2" % "provided",
+      compilerPlugin("org.typelevel"   %% "kind-projector"  % "0.10.3"),
+      compilerPlugin("com.github.ghik" %% "silencer-plugin" % "1.4.2")
     )
+    CrossVersion.partialVersion(scalaVersion) match {
+      case Some((2, x)) if x <= 12 =>
+        stdCompileOnlyDeps ++ Seq(
+          compilerPlugin(("org.scalamacros" % "paradise" % "2.1.1").cross(CrossVersion.full))
+        )
+      case _ => stdCompileOnlyDeps
+    }
   }
 
-  def extraOptions(scalaVersion: String) =
+  private def compilerOptions(scalaVersion: String, optimize: Boolean) = {
```

Comment:
When would we need to actually use `optimize = false`?

Reply:
The idea is to only use these options when building an actual release.
opt: l: inline for example, is quite broken otherwise AFAIK https://github.com/sbt/zinc/issues/537

Reply:
I see, I was not aware of this issue :+1: